packages/app/src/components/Table/Table.tsx

```diff
@@ -118,76 +139,94 @@ const Table = <T,>({
+														<StyledSortUpIcon />
+													)
 												) : (
-													<StyledSortUpIcon />
-												)
-											) : (
-												<>
-													<StyledSortUpIcon />
-													<StyledSortDownIcon />
-												</>
-											)}
-										</SortIconContainer>
-									)}
-								</TableCellHead>
-							)
-						})}
-					</FlexDiv>
-				))}
-				{isLoading ? (
-					<Loader />
-				) : !!noResultsMessage && data.length === 0 ? (
-					noResultsMessage
-				) : (
-					<TableBody className="table-body">
-						{table.getRowModel().rows.map((row, idx) => {
-							const localRef =
-								lastRef && idx === table.getState().pagination.pageSize - 1 ? lastRef : defaultRef
-							return (
-								<TableBodyRow
-									key={row.id}
-									localRef={localRef}
-									highlightRowsOnHover={highlightRowsOnHover}
-									row={row}
-									onClick={handleRowClick(row)}
-								/>
-							)
-						})}
-					</TableBody>
-				)}
-				{showPagination && data.length > table.getState().pagination.pageSize ? (
-					<Pagination
-						compact={compactPagination}
-						pageIndex={table.getState().pagination.pageIndex}
-						pageCount={table.getPageCount()}
-						canNextPage={table.getCanNextPage()}
-						canPreviousPage={table.getCanPreviousPage()}
-						setPage={table.setPageIndex}
-						previousPage={table.previousPage}
-						nextPage={table.nextPage}
-					/>
-				) : undefined}
-			</ReactTable>
-		</TableContainer>
+													<>
+														<StyledSortUpIcon />
+														<StyledSortDownIcon />
+													</>
+												)}
+											</SortIconContainer>
+										)}
+									</TableCellHead>
+								)
+							})}
+						</FlexDiv>
+					))}
+					{isLoading ? (
+						<Loader />
+					) : !!noResultsMessage && data.length === 0 ? (
+						noResultsMessage
+					) : (
+						<TableBody className="table-body">
+							{table.getRowModel().rows.map((row, idx) => {
+								const localRef =
+									lastRef && idx === table.getState().pagination.pageSize - 1 ? lastRef : defaultRef
+								return (
+									<TableBodyRow
+										key={row.id}
+										localRef={localRef}
+										highlightRowsOnHover={highlightRowsOnHover}
+										row={row}
+										onClick={handleRowClick(row)}
+									/>
+								)
+							})}
+						</TableBody>
+					)}
+					{(shouldShowPagination || paginationExtra) && paginationVariant !== 'staking' ? (
+						<Pagination
+							compact={compactPagination}
+							pageIndex={table.getState().pagination.pageIndex}
+							pageCount={table.getPageCount()}
+							canNextPage={table.getCanNextPage()}
+							canPreviousPage={table.getCanPreviousPage()}
+							setPage={table.setPageIndex}
+							previousPage={table.previousPage}
+							nextPage={table.nextPage}
+							variant={paginationVariant}
+							extra={paginationExtra}
+						/>
+					) : undefined}
+				</ReactTable>
+			</TableContainer>
+			{paginationVariant === 'staking' && (
```

Comment:
We shouldn't really be including parent specific logic inside here. Would be good if we can have a solution to keep this component generic.

Reply:
Thanks. Updated the code according to #2583